lib/ews/soap/exchange_notification.rb

```diff
@@ -63,7 +65,16 @@ def subscribe(subscriptions)
           }
         end
       end
-      do_soap_request(req, response_class: EwsResponse)
+      do_soap_request(req, response_class: EwsResponse, request_options: request_options)
+    end
+
+    def get_request_options(options)
```

Comment:
is this method to select only customisable headers and cookies?  could we use `#reject` instead?